trie.go

```diff
@@ -89,22 +89,33 @@ func (t *Trie) HasKeysWithPrefix(key string) bool {
 
 // Removes a key from the trie, ensuring that
 // all bitmasks up to root are appropriately recalculated.
-func (t *Trie) Remove(key string) {
+func (t *Trie) Remove(key string) bool {
```

Comment:
I don't think we should change the function signature here as it would break consumers of the library. Any breaking change would require a major version update.